sync/src/test/java/tech/pegasys/artemis/sync/BlockPropagationManagerTest.java

```diff
@@ -281,7 +284,8 @@ public void onBlockImported_withPendingFutureBlocks() throws Exception {
   private UnsignedLong incrementSlot() {
     currentSlot = currentSlot.plus(UnsignedLong.ONE);
     localChain.setSlot(currentSlot);
-    localEventBus.post(new SlotEvent(currentSlot));
+    blockPropagationManager.onSlot(new SlotEvent(currentSlot));
+    blockPropagationManager.pendingBlocks.onSlot(new SlotEvent(currentSlot));
```

Comment:
I'd be tempted to use a real (sync) `EventChannels` rather than a mock, then you can just call `eventChannels.getPublisher(SlotEventsChannel.class).onSlot(...)`.